src/goto-checker/report_util.cpp

```diff
@@ -181,69 +181,80 @@ static void output_single_property_plain(
                << messaget::eom;
 }
 
+using propertyt = std::pair<irep_idt, property_infot>;
+/// Compare two properties according to the following sort:
+/// 1. alphabetical ordering of file name
+/// 2. alphabetical ordering of function name
+/// 3. numerical ordering of line number
+/// 4. alphabetical ordering of goal ID
+/// 5. number ordering of the goal ID number
+/// \param property1: The first property.
+/// \param property2: The second propery.
+/// \return True if the first property is less than the second property
+static bool
+is_property_less_than(const propertyt &property1, const propertyt &property2)
+{
+  const auto &p1 = property1.second.pc->source_location;
+  const auto &p2 = property2.second.pc->source_location;
+  if(p1.get_file() != p2.get_file())
+    return id2string(p1.get_file()) < id2string(p2.get_file());
+  if(p1.get_function() != p2.get_function())
+    return id2string(p1.get_function()) < id2string(p2.get_function());
+  else if(
+    !p1.get_line().empty() && !p2.get_line().empty() &&
+    p1.get_line() != p2.get_line())
+    return std::stoul(id2string(p1.get_line())) <
+           std::stoul(id2string(p2.get_line()));
+
+  const auto split_property_id =
+    [](const irep_idt &property_id) -> std::pair<std::string, std::size_t> {
+    const auto property_string = id2string(property_id);
+    const auto last_dot = property_string.rfind('.');
+    std::string property_name;
+    std::string property_number;
+    if(last_dot == std::string::npos)
+    {
+      property_name = "";
+      property_number = property_string;
+    }
+    else
+    {
+      property_name = property_string.substr(0, last_dot);
+      property_number = property_string.substr(last_dot + 1);
+    }
+    const auto maybe_number = string2optional_size_t(property_number);
+    if(maybe_number.has_value())
+      return std::make_pair(property_name, *maybe_number);
+    else
+      return std::make_pair(property_name, 0);
+  };
+
+  const auto left_split = split_property_id(property1.first);
+  const auto left_id_name = left_split.first;
+  const auto left_id_number = left_split.second;
+
+  const auto right_split = split_property_id(property2.first);
+  const auto right_id_name = right_split.first;
+  const auto right_id_number = right_split.second;
+
+  if(left_id_name != right_id_name)
```

Comment:
FYI this is the same as `std::tie(left_id_name, left_id_number) < std::tie(right_id_name, right_id_number)`.